components/CopyComponent.tsx

```diff
@@ -0,0 +1,61 @@
+import { Pressable, StyleSheet } from "react-native";
+import { ThemedText } from "./ThemedText";
+import { Ionicons } from "@expo/vector-icons";
+import { Colors } from "@/constants/Theme";
+import * as Clipboard from "expo-clipboard";
+import { useState } from "react";
+
+export function CopyComponent({ gameCode }: { gameCode: string }) {
+  const [copied, setCopied] = useState(false);
+
+  const copyToClipboard = () => {
+    Clipboard.setStringAsync(gameCode);
+    setCopied(true);
+    setTimeout(() => setCopied(false), 2000);
+  };
+
+  return (
+    <>
+      <Pressable style={styles.button} onPress={copyToClipboard}>
+        {copied ? (
+          <ThemedText style={styles.buttonText} type="defaultSemiBold">
+            Copied!
+          </ThemedText>
+        ) : (
+          <ThemedText style={styles.buttonText} type="defaultSemiBold">
+            {gameCode}
+          </ThemedText>
+        )}
+
+        <Ionicons
+          style={styles.copy}
+          name={copied ? "checkmark-outline" : "copy-outline"}
+          size={18}
+          color={Colors.light.addButton}
+        />
+      </Pressable>
+    </>
+  );
+}
+
+const styles = StyleSheet.create({
+  button: {
+    backgroundColor: Colors.light.contrastBlue,
+    padding: 10,
```

Comment:
Nu har jag i och för sig inte lagt in så mycket i theme än, men dessa finns i alla fall att använda istället för 10, 15 etc. :) 
Spacings: {
    xSmall: 5,
    small: 10,
    medium: 15,
    large: 20,
    xLarge: 25,
  },